docs/code-standards-policy.rst

```diff
@@ -118,36 +168,483 @@ using the debops.apt_install_ role MUST be defined as (or similar):
+
+Whenever possible DebOps role authors MUST specify role dependencies via
+playbook instead of
+:ref:`"hard" dependencies <debops_policy__ref_code_standards_hard_role_dependencies>`
+in the :file:`meta/main.yml`.
+
+It's also possible to pass configuration values to a dependent role if the
+involved role is offering dedicated
+:ref:`dependent variables <debops_policy__ref_code_standards_dependent_variables>`
+for this purpose. With soft dependencies custom playbook authors are therefore
+free to pass values from arbitrary sources according to their requirements.
+
+**Example:**
+
+This is an example playbook for debops.slapd_ defining soft dependencies:
+
+.. code-block:: yaml
+
+    ---
+
+    - name: Manage OpenLDAP service
+      hosts: [ 'debops_service_slapd' ]
+      become: True
+
+      roles:
+
+        - role: debops.ferm
+          tags: [ 'role::ferm' ]
+          ferm__dependent_rules:
+            - '{{ slapd__ferm__dependent_rules }}'
+
+        - role: debops.tcpwrappers
+          tags: [ 'role::tcpwrappers' ]
+          tcpwrappers__dependent_allow:
+            - '{{ slapd__tcpwrappers__dependent_allow }}'
+
+        - role: debops.slapd
+          tags: [ 'role::slapd' ]
+
+
+.. _debops_policy__ref_code_standards_hard_role_dependencies:
+
+Hard role dependencies
+~~~~~~~~~~~~~~~~~~~~~~
+
+Role dependencies are considered "hard" if they are defined in the
+``dependencies`` list in :file:`meta/main.yml`.  DebOps role authors MUST
+avoid the use of hard role dependencies for the follwing reasons:
+
+- Hard role dependencies must always be installed on the Ansible controller
+  even when their execution is conditionally triggered via ``when`` statement.
+
+- It hinders the independent use of the role in a custom playbook or outside
+  of DebOps where playbook authors might relay on a different role for a certain
+  feature or decide no to use a certain feature at all.
+
+- The playbook execution flow is more difficult to reason about as hard
+  dependencies are defined outside of the playbook.
+
+Generally role dependencies MUST be defined as
+:ref:`"soft" dependencies <debops_policy__ref_code_standards_soft_role_dependencies>`
+via playbook unless the tight coupling to another role is unavoidable for
+implementing the required functionality. A reasonable exception is for example
+the debops.secret_ role which defines a common path for the
+``lookup("password")`` module.
+
+
+.. _debops_policy__ref_code_standards_role_facts:
+
+Ansible role facts
+------------------
+
+Ansible facts are small things that are automatically discovered by the
+`Ansible setup module`_ on a target host when a playbook is executed. They can
+be used by Ansible role and playbook authors through normal variables. The
+default facts provided are indicated by the ``ansible_`` namespace. It's
+possible to define custom facts through JSON or INI files or scripts returning
+such output in the :path:`/etc/ansible/facts.d` directory of a host. These
```

Comment:
I don't think that `:path:` is a valid Spinx err, statement? Use `:file:` instead.